src/hooks/useGetPledgeOrDonation.ts

```diff
@@ -0,0 +1,183 @@
+import { useEffect, useState } from 'react';
+import { TFunction } from 'i18next';
+import { DateTime } from 'luxon';
+import { useTranslation } from 'react-i18next';
+import { AppealStatusEnum } from 'src/components/Tool/Appeal/AppealsContext/AppealsContext';
+import { AppealContactInfoFragment } from 'src/components/Tool/Appeal/AppealsContext/contacts.generated';
+import { PledgeFrequencyEnum } from 'src/graphql/types.generated';
+import { currencyFormat, dateFormat } from 'src/lib/intlFormat';
+import { getLocalizedPledgeFrequency } from 'src/utils/functions/getLocalizedPledgeFrequency';
+
+type FormatPledgeOrDonationProps = {
+  amount?: number | null;
+  currency?: string | null;
+  appealStatus: AppealStatusEnum;
+  dateOrFrequency?: PledgeFrequencyEnum | string | null;
+  locale: string;
+  t: TFunction;
+};
+
+const formatPledgeOrDonation = ({
+  amount,
+  currency,
+  appealStatus,
+  dateOrFrequency,
+  locale,
+  t,
+}: FormatPledgeOrDonationProps) => {
+  const pledgeOrDonationAmount =
+    amount && currency
+      ? currencyFormat(amount, currency, locale)
+      : amount || currencyFormat(0, currency, locale);
+
+  let pledgeOverdue = false;
+  if (
+    (appealStatus === AppealStatusEnum.NotReceived ||
+      appealStatus === AppealStatusEnum.ReceivedNotProcessed) &&
+    dateOrFrequency
+  ) {
+    const date = DateTime.fromISO(dateOrFrequency).startOf('day');
+    if (date <= DateTime.local().startOf('day')) {
+      pledgeOverdue = true;
+    }
+  }
+
+  const pledgeOrDonationDate =
+    appealStatus === AppealStatusEnum.Asked ||
+    appealStatus === AppealStatusEnum.Excluded
+      ? (dateOrFrequency &&
+          getLocalizedPledgeFrequency(
+            t,
+            dateOrFrequency as PledgeFrequencyEnum,
+          )) ??
+        ''
+      : dateOrFrequency
+      ? dateFormat(DateTime.fromISO(dateOrFrequency), locale)
+      : null;
+  return {
+    amount: pledgeOrDonationAmount,
+    dateOrFrequency: pledgeOrDonationDate,
+    pledgeOverdue,
+  };
+};
+
+export interface UseGetPledgeOrDonation {
+  pledgeValues: AppealContactInfoFragment['pledges'][0] | undefined;
+  amountAndFrequency: string[] | undefined;
```

Comment:
Thoughts on making this type `[string] | [string, string] | undefined` to enforce the fact that it should only contain one or two values? While we're at it, is there a reason it isn't `{ amount: string, frequency?: string }`? That would be even more self-documenting for future-devs. Sorry I didn't think of that the first time I reviewed this logic.